lib/mrb/scripts/command_line/grndb.rb

```diff
@@ -402,6 +405,37 @@ def failed_to_open(name)
           failed(message)
         end
       end
+
+      class Recoverer
+        attr_writer :database
+
+        def initialize
+          @context = Context.instance
+        end
+
+        def recover(force_truncate)
+          @database.recover
+          if force_truncate
+            @database.each do |object|
+              if object.corrupt?
```

Comment:
We should use 'next unless object.corrupt?` instead of deep nested code.
See also: http://www.clear-code.com/blog/2012/3/28.html (in Japanese)